apps/web/lib/video/[uid]/getServerSideProps.ts

```diff
@@ -59,6 +60,21 @@ const shouldEnableAutomaticTranscription = ({
   return !!calVideoSettings.enableAutomaticTranscription;
 };
 
+const shouldEnableAutomaticRecording = ({
+  hasTeamPlan,
+  calVideoSettings,
+  isOrganizer,
+}: {
+  hasTeamPlan: boolean;
+  calVideoSettings?: CalVideoSettings | null;
+  isOrganizer: boolean;
+}) => {
+  if (!hasTeamPlan || !isOrganizer) return false;
```

Comment:
Why do we need to check for hasTeamPlan separately if calVideoSettings is guiding the feature directly.

calVideoSetting must have had the value set after validating all the things.